### DISCUSSION TITLE
yolo 모델 학습 여부, 버전 업그레이드 논의

### DISCUSSION THREAD

**Discussion by kwakmoonjung** [MechanIT/Speech-and-object-recognition-for-elderly-people-living-alone#8] [Ideas]
yolo 모델을 직접 학습시켜보는 게 어떨까요?

원래는 저희가 누워 있는 사람을 구분할 수 있는 FPDS 데이터셋을 사용하기로 했었습니다.
FPDS dataset 제작자가 이미 이 dataset을 yolov3 모델로 학습시켜놓은 weight 파일 (darknet 기반)을 업로드해놨습니다.
따라서 이 weight 파일으로  yolo 모델을 구동해보려고 했는데, 계속 에러가 발생했습니다.

- 가은님 시도 : darknet weight 파일을 tensorflow 기반 weight 로 변환
    -> bounding box는 만들어지나, 오인식 (사람을 고양이라고 인식)

- 제 시도 : darknet 설치하여 실행 시도
    -> darknet 파일 빌드 에러 발생

이미 학습되어 있는 weight 파일로 객체 인식을 진행하기보다는, 
교수님께서 피드백해주셨던 것처럼 **저희가 직접 데이터셋을 만들고, 학습까지 해서 weight 파일을 직접 만들어 보는 건 어떨까요?**
**이왕 학습시키는 김에, yolo 모델 버전을 높여서 실시간성도 높여보면 좋을 것 같아요.**
yolo 모델 버전을 높일수록 환경 세팅이 까다롭고 어렵긴 하지만, local에서 진행하지 않고 colab에서 모델을 학습시키면 간단하게 환경설정 가능한 것으로 확인했습니다.

**Comment by GaeunLee1120** [comment]
### - yolov3 weight 시도
darknet 호환 이슈로 darknet weight 파일을 tensorflow weight로 변환 후 시도해봤습니다. 
변환 후 streaming 화면을 출력하고, bounding box가 나오는 것까지는 확인했으나

**bounding box 위치와 판단이 정확하지 않습니다.**

> 1. 사람을 bounding box로 인지하지 못함
> 2. 사람을 sofa로 인식함 (단순히 sofa만으로만 인식하지 않고 cat, 등등 경우에 따라 다르게 인식함)
> 
![yolov3](https://github.com/MechanIT/Speech-and-object-recognition-for-elderly-people-living-alone/assets/138747086/a285ef02-3ee9-4266-a65a-225023b38e1e)


저도 **버전을 업그레이드 후 저희의 사진을 직접 찍어 학습을 시키는 것이 좋을 것 같습니다.** 
우선, yolo 버전을 업그레이드 후 성능 차이를 확인하고, (yolov5) 사람의 자세를 따로 구별을 하는 지 확인해보려 합니다.

**Reply by kwakmoonjung** [reply]
그럼 저는 yolov7 학습 방법을 조사해보고, 학습 진행해보도록 하겠습니다.

**Reply by GaeunLee1120** [reply]
yolov5로 사람 인식해본 결과, v3와 다르게 누워있는 사람도 person으로 인식하는 것을 확인했습니다.
(그러나, 일자로 서있는 상태보다는 정확도가 떨어지는 것 같습니다.)
![image](https://github.com/MechanIT/Speech-and-object-recognition-for-elderly-people-living-alone/assets/138747086/0a5ca17c-ff56-4ae1-b245-6e9c9daef70a)


그러나, 저희가 필요한 fallen person / person 객체는 구별 기능은 되지 않아서, 학습을 진행해보려 합니다.
가장 최근에 나온 모델이 **yolov9**이라 간단한 기능 테스트 후, **학습 진행**해보도록 하겠습니다.
